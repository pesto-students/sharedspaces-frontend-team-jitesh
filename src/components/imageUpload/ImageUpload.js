import Axios from "../../Axios";
import React from "react";
import { useState } from "react";
import "./imageUpload.scss"
import Loader from "../loader/Loader";

const ImageUpload = ({
    uploadType,
    id,
    uploadPath
}) => {
    const [file, setFile] = useState();
    const [loading, setLoading] = useState(false)

    const handleChange = (event) => {
        onUpload(event.target.files[0]);
        setFile(URL.createObjectURL(event.target.files[0]));
    };


    const guidGenerator = () => {
        var S4 = function () {
            return (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
        };
        return (S4() + S4() + "-" + S4() + "-" + S4() + "-" + S4() + "-" + S4() + S4() + S4());
    }

    const onUpload = async (file) => {

        const formData = new FormData()
        formData.append('file', file)

        try {
            setLoading(true)
            const { data } = await Axios.post(`/image/${uploadType}/${id ? id : guidGenerator()}`, formData)
            if (data.success) {
                setFile(data.data)
                setLoading(false)
                uploadPath(data.data)
            } else {
                setLoading(false)
            }
        } catch (error) {
            setLoading(false)
        }
    }

    return (
        <div className="upload-wrapper shadow-new rounded">
            {loading ?
                <Loader width={"w-10"} className="text-gray-200" /> :
                <label htmlFor="featured">
                    {file ? (
                        <img className="uploaded-image rounded" src={file} alt="" />
                    ) : (
                        <img
                            className="uploaded-image rounded"
                            src={"/assets/images/image-upload.png"
                            }
                            alt=""
                        />
                    )}
                </label>
            }
            <input
                id="featured"
                type="file"
                className="invisible"
                onChange={handleChange}
            />
        </div>
    );
};

export default ImageUpload;
