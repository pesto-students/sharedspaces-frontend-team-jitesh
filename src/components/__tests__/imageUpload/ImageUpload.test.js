import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer'
import ImageUpload from '../../imageUpload/ImageUpload'

afterEach(() => {
  cleanup()
})

test('test matches snapshot', () => {
  const component = renderer.create(<ImageUpload />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot();
})

test('should render imageUpload component', () => {
  render(<ImageUpload />)

  const imageUploadElement = screen.getByTestId("imageUpload")

  expect(imageUploadElement).toBeInTheDocument();
})
