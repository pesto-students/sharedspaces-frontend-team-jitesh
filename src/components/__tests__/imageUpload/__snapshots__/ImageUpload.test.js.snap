// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<div
  className="upload-wrapper shadow-new rounded"
  data-testid="imageUpload"
>
  <label
    htmlFor="featured"
  >
    <img
      alt=""
      className="uploaded-image rounded"
      src="/assets/images/image-upload.png"
    />
  </label>
  <input
    className="invisible"
    id="featured"
    onChange={[Function]}
    type="file"
  />
</div>
`;
